modules/ROOT/pages/addition/agent-installation/self-registered.adoc

```diff
@@ -0,0 +1,169 @@
+:description: this page describes NOM agent self-registering setup.
+
+Agent self registration omits manually aquiring initial configuration from NOM UI as described in xref:./manual.adoc#register[Manually registering agent] and instead relies on additional agent configuration on Neo4j instance host to obtain initial configuration from NOM server either through mutual authentication or manual authorization of the agent from UI after installation.
+
+A self-registering agent is useful in an automated environment such as a Kubernetes cluster of Neo4j instances where instances are dynamically created and managed.
+
+[[configure]]
+== Configuration for agent self registration with manual approval
+Following are the required environment variables to be configured to enable self-registration of NOM agent:
+
+[cols="<,<,<",options="header"]
+|===
+| Variable
+| Description
+| Example
+
+| `CONFIG_SERVER_GRPC_ADDRESS`
+| Server GRPC Address
+| server:9090
+
+| `CONFIG_SERVER_HTTP_ADDRESS`
+| Server HTTP address
+| +++https://server:8080+++
+|===
+
+With only the above configuration, agent attempts to connect to NOM server for self-registration which results in agent going into *Standby mode* until authorized.
+
+IMPORTANT: *Standby mode* - this describes the runtime state of a self registering agent which is not authorized yet to monitor an instance. In this mode agent keeps checking with NOM server for approval status at every preset time interval (__default 30 seconds__).
+
+An agent in standby mode shows up in the list of agents in NOM UI. To enable the agent to continue it's normal execution, agent needs to be approved from NOM UI as shown below:
+
+image::agents.png[width=800]
+
+To approve the agent click on `...` agent action icon and click `Approve`:
+
+image::agents.png[width=800]
+
+An edit view to optionally update agent name or description will show as below:
+
+NOTE: an unauthorized self-registering agent will have default name derived from it's host information. It's recommended to update this name or use the optional configuration described <<agent-meta-data,here>>.
+
+image::agent-new.png[width=800]
+
+Once the agent is approved, by optionally updating it's name and description, agent will receive initial configuration from NOM server which it persists on the instance host in a file named `nom-agent-config.yaml`.
+
+WARNING: Location of where agent saves it's initial configuration must be of persistent type else agent needs to be reregistered.
+
+Following optional configuration can be used to provide a persistent location for agent intial configuration:
+
+[cols="<,<,<,<",options="header"]
+|===
+| Variable
+| Description
+| Example
+| Default
+
+| `CONFIG_AGENT_CONFIG_PATH`
+| Peristent path to a file on Neo4j instance host
+| "file://path/to/"
+| by default agent tries to save initial configuration in path `NEO4J_CONF` if set or `conf` folder under `NEO4J_HOME` if set, else the file is saved under `.nom` folder in user home directory.
```

Comment:
What if no instances are configured? 
What if multiple instances are configured?

Reply:
No instances would result in config being saved to user home directory. for multiple instances, since agent looks for current env for config folder path, it could only be one value or no value per host right?